### ISSUE TITLE
"Plots" pane on Windows not showing ggplot2::geom_line() when it contains an alpha

### ISSUE THREAD

**Issue by daattali** [rstudio/rstudio#2196]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.1.414
    OS Version      : Windows 7
    R Version       : 3.4.3

This was [initially reported as a ggplot2 bug](https://github.com/tidyverse/ggplot2/issues/2452) but was closed due to being likely a graphics device issue. It may not be an IDE issue, but I wanted to post it here in case there *is* something that can be done about it from the IDE end. This issue happens on Windows only.

The following code results in a plot with a legend but the lines in the legend disappear

```
ggplot(mtcars, aes(wt, colour = factor(vs))) + geom_line(stat="density", alpha = 0.9)
```

![image](https://user-images.githubusercontent.com/952340/36180545-ec646b50-10ee-11e8-9f30-fbb5dd2b3394.png)

If I click on the "Zoom" button and it pops the plot into its own window, then the lines in the legend appear. Making the plot pane much larger does not seem to make a difference.

A more serious version of this bug causes the lines to not show up at all. The following code:

```
ggplot(mtcars, aes(wt, colour = factor(vs))) + geom_line(stat = "density", alpha = 0.9) +
  theme(
    axis.title.x = element_blank(),
    axis.text.x = element_blank(),
    plot.margin = unit(c(0, 0, 0, 0), "null"),
    axis.ticks.length = unit(0, "null")
  )
```

(I just added a few theme parameters) results in this plot in the Plot pane

![image](https://user-images.githubusercontent.com/952340/36180895-502ea28a-10f0-11e8-9821-8b1bb805841b.png)

Removing any of the theme parameters will make the plot appear correctly, or using the Zoom button.

**Comment by yutannihilation** [comment]
Curious issue... For comparison, it seems Rterm can display it correctly:

![image](https://user-images.githubusercontent.com/1978793/36211276-25e001be-11e4-11e8-9c41-2634c1d668c4.png)

**Comment by dfalty** [comment]
I am unable to reproduce the first issue on Win7 and Win8 but the second example reproduces on both (1.1.423). Thanks for the report!

**Comment by dfalty** [comment]
@yutannihilation You are right, I'm able to reproduce this as well! (Win8 1.1.423) I somehow missed the first case was restricted to the legend. Thanks for the follow-up and clarification.

**Comment by yutannihilation** [comment]
@dfalty Thanks for confirming!

**Comment by jjallaire** [comment]
I want to give a word of warning about fixing this issue. Our graphics device implementation is pretty complex (creating a "shadow" PNG device in the background which we delegate to and periodically flush). I once tried to make what I thought was a straightforward change to support transparent backgrounds: 

https://github.com/rstudio/rstudio/commit/194e20d1a8b03e7249fb671d89a3ee405d1053f3#diff-99a7469c2717e27e2be7ca21e9494f7d

Then had to revert it shortly thereafter as it broke multiple other things:

https://github.com/rstudio/rstudio/commit/4f906392a56493aae476e38f2fd53552ed58ff44#diff-99a7469c2717e27e2be7ca21e9494f7d

Basically, any changes to our graphics device implementation are likely to cause other downstream problems so the issues they address need to severe to justify the potential breakage.

**Comment by jmcphers** [comment]
As @jjallaire notes the expense and risk of fixing this is high, and the severity is relatively low (there have been no reports in well over a year). For this reason it's unlikely to make a release in the next year or two.

**Comment by daattali** [comment]
@jmcphers for what it's worth, I'm still seeing this bug happen, even on Windows 10.

**Comment by vankesteren** [comment]
Same, with the right margin set to 0 in the following `ggplot` code:
```r
library(ggplot2)
df <- data.frame(
  time = rep(1:10, 100), 
  val  = rnorm(1000), 
  id   = rep(1:100, each = 10)
)
ggplot(df, aes(x = time, y = val, group = id)) + 
  geom_point() +
  geom_line(alpha = 0.1) +
  theme(plot.margin = margin(1, 0, 1, 1))
```

**Comment by jmcphers** [comment]
Reopening since it seems to keep coming up on the ggplot2 side.

**Comment by mehrlander** [comment]
I just realized this was the problem after fumbling around for an hour to get my legend lines to show. I have Windows 10 and the RStudio 1.3.959.  It seems like this must be common if it's any time you have lines and alpha.  I used "guides(alpha = guide_legend(override.aes = list(size = 4)))" which successfully made it thicker in the "zoom" popup window, but didn't make it show up in the plots pane.

**Comment by yutannihilation** [comment]
I guess this issue is virtually closed by this PR, though we need to wait for v1.4.
https://github.com/rstudio/rstudio/pull/6539

**Comment by yutannihilation** [comment]
As RStudio v1.4 is released so the users now can use AGG to avoid this problem, maybe we can close this issue?

**Comment by daattali** [comment]
I just tested using RStudio version 1.4.1103 (windows 10)

I'm still seeing the same issue. The following code results in a completely blank plot:

```
library(ggplot2)
ggplot(mtcars, aes(wt, colour = factor(vs))) + geom_line(stat = "density", alpha = 0.9) +
  theme(
    axis.title.x = element_blank(),
    axis.text.x = element_blank(),
    plot.margin = unit(c(0, 0, 0, 0), "null"),
    axis.ticks.length = unit(0, "null")
  )
```

**Comment by yutannihilation** [comment]
Really? Did you install ragg package and turn on the setting...?

> To start using this new device, go to Options -> General -> Graphics -> Backend and select “AGG”.
https://blog.rstudio.com/2020/12/02/rstudio-v1-4-preview-little-things/

**Comment by daattali** [comment]
I did not, I didn't know what AGG means, I assumed that was something ubilt in by default. 

You're right, it works with AGG turned on. The blog post doesn't mention is there's any drawbacks to using AGG, is there a reason it's not default?

**Comment by yutannihilation** [comment]
Yeah, agreed. I thought this was going to be default...

https://github.com/rstudio/rstudio/issues/8437 seems to try solving this in a bit different direction, which might take some more time?

> We'd actually take the native code from ragg as opposed to depending on the package.

**Comment by jmcphers** [comment]
> The blog post doesn't mention is there's any drawbacks to using AGG, is there a reason it's not default? 

It's mainly because you need to install additional software (specifically the ragg R package) to make it work seamlessly with the IDE.

**Comment by daattali** [comment]
I see. And it looks like {ragg} does have a lot of system requirements, so I understand not wanting to make it default. As long as we can install it, it's better and has no drawbacks over the default device?

@jmcphers perhaps having a small informational button inside the "Plots" pane would be helpful, maybe a cog that would take you to the Graphics settings? I know this is hijacking this thread a bit but I don't want to open a new issue for it sine I know you're trying to cut down on issues, just wanted to make a one time suggestion and you can decide if it's worth following up :)

![image](https://user-images.githubusercontent.com/952340/108637163-70e58980-7457-11eb-8128-926bc24fbb2c.png)

**Comment by yutannihilation** [comment]
I'm not sure if RStudio needs its own fix, but this is the fix committed to r-devel.

https://github.com/wch/r-source/commit/3c38741b10d81c3a3faf17d1902901ae18454e11
